docs/compatibility_draft.md

```diff
@@ -0,0 +1,246 @@
+- 1/7/23: Add enum `<B>` which has new semantics in libStablehlo@v2
+  + No upgrade needed since `<A>` exists in both builds.
+  + Add bytecode verifier pass with `emitWarning` if `<B>` is used, warning of forwards incompatibility.
+- 1/29/23: Remove warning in libStablehlo@v3
+  + Forward and backward compatibility fully supported after 3 weeks.
+
+### Deprecating a Feature
+_Characteristics of a deprecation change:_
+- Remove a feature with backwards compatibility
+- Prevent creation of new bytecode using feature
+- Clean up after backward compatibility window
+
+_Example:_ Removing an op / attribute / enum.
+
+_Compatibility Protocol:_
+
+<img width="951" alt="deprecate" src="https://user-images.githubusercontent.com/3036970/189246238-96ef5fdd-a735-40ef-982d-c4a1b5cc114a.png">
+
+_Description:_
+- 1/1/23: Create bytecode_v1 from libStablehlo@v1
+- 1/7/23: Add verifier to prevent serialization of `<B>` in libStablehlo@v2
+  + Do not prevent the deserialization of `<B>`
+  + No change in serialization of `<A>` so still forward compatible
+- 6/8/23: Remove `<B>` as enum value
+  + Backward compatibility window over.
+  + Ensure that numeric values of other enums do not change.
+
+### Versioning an Operation
+_Characteristics of a change that requires versioning:_
+- A change is introduced that cannot be upgraded / downgraded
+- Create new version of legacy op
+- A combination of Addition/Deprecation Protocols from there
+
+_Example:_ Changing the signature and semantics to an op, adding an operand.
+
+_Compatibility Protocol:_
+
+<img width="941" alt="version" src="https://user-images.githubusercontent.com/3036970/189246254-c13e6509-2ef4-4b4f-a56d-186c4198e2b8.png">
+
+_Description:_
+- 1/1/23: Create `bytecode_v1` from libStablehlo@v1
+- 1/7/23: Add verifier to prevent serialization of `MyOp` in libStablehlo@v2
+  + Add verifier to warn if `MyOp_v2` is used and cannot be downgraded.
+  + No change in serialization so still forward compatible
+- 1/29/23: Remove forward compatibility warning 
+- 6/8/23: Remove `MyOp` in libStablehlo@v4
+  + Backward compatibility window over.
+  + Optionally rename `MyOp_v2` to `MyOp` via Migration Protocol
+
+### Changes in Dependent Dialects
+StableHLO will use forks of supported operations in dependent dialects for serialization / deserialization only. These forked operations allow StableHLO a finer control of the compatibility changes of operations in other dialects using the protocols above.
+
+_Proposal:_
+- Have stable shims of supported operations in other dialects.
+- Downgrade IR to shim before bytecoding `arith.addi → stablehlo.arith.addi`
+- Upgrade IR to other dialect on deserialization.
+
+With the above rules, we are able to provide compatibility guarantees for Migration and Additions. 
+
+_Example:_ Move `builtin.func → func.func`.
+
+<img width="943" alt="dependent" src="https://user-images.githubusercontent.com/3036970/189246267-06991d87-71a7-4ea8-bca0-5c818b0f2964.png">
```

Comment:
`stable.op` in this picture probably means `stablehlo.op`?

Reply:
Fixed